### ISSUE TITLE
Reactor previews do not re-render when handlers change

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#1641]
See short demo below:

![Mar-09-2022 09-21-45](https://user-images.githubusercontent.com/119891/157496173-51513181-ed23-4561-9fe5-e47faefc0736.gif)

**Comment by shriram** [comment]
This raises a semantic question: since a reactor is a time-varying value, should the preview always show the latest value of the reactor? Does this give us FrTime through the back door? (-:

**Comment by blerner** [comment]
No - because reactors are not time-varying unless you `.interact` with them.  Joe's screenshot above does seem like a bug, without requiring fully-FRP values in our REPL.

**Comment by shriram** [comment]
(Actually, the above isn't even showing the init state, it's showing the `to-draw` of the init I believe…

**Comment by jpolitz** [comment]
The preview right now the result of `<reactor>.draw()`. I want to add more gadgetry around this with an explicit “play” button to set it apart; right now it's indistinguishable from an image (which @schanzer reminded me of) and that's bad.

**Comment by blerner** [comment]
Yeah -- was mulling this over yesterday a bit: I'd like to see a toggleable UI similar to our fractions, that would flip between rendering the world state, and rendering the `.draw()` output of the world state...

**Comment by shriram** [comment]
Oooh, I love this! This would really drive home something about the essence
of reactors.

**Comment by asolove** [comment]
Is there currently a way to work with reactors in Anchor? It looks like the reactor itself now renders as a representation of the object, draw shows the `to-draw(init)`, and calling `interact` loops forever (as if the simulation was running internally without displaying anything in the UI).

<img width="644" alt="image" src="https://user-images.githubusercontent.com/8495/210161017-b2230392-312c-4269-9bbc-fe8fb15c97e4.png">

**Comment by CosineP** [comment]
Click the image :)   As Joe mentioned above, that's not made clear at all.   Calling interact should also probably do something clever

**Comment by asolove** [comment]
@CosineP as far as I can tell, on [the currently-published version of anchor](https://code.pyret.org/anchor/?program=include%20cpo%0Ainclude%20image%0Ainclude%20reactors%23.CHUNK%23%0Afun%20on-tick(x%20%3A%3A%20Number)%3A%0A%20%20x%20%2B%200.1%0Aend%23.CHUNK%23%0Afun%20to-draw(n%20%3A%3A%20Number)%3A%20circle(n%2C%20%22solid%22%2C%20%22blue%22)%20end%23.CHUNK%23%0Ar%20%3D%20reactor%3A%20init%3A%2025%2C%20on-tick%3A%20on-tick%2C%20to-draw%3A%20to-draw%20end%23.CHUNK%23%0Ar.draw()), that doesn't work. The reactor definition now shows the object properties view of the object rather than its draw, and calling `draw` on it shows a static image with no over-time behavior. Neither of them do anything on click, hover, etc.

**Comment by CosineP** [comment]
Ah, right, the current version is missing that.  In your picture, where you see `r = ...`, that should render to-draw(init) like you've manually done below, and clicking it should open a pop-up window that interacts (like in the video in the first message), but IIRC the value type descrimination got rewritten and doesn't include reactors now.   If you want to see what's up, the old logic is in reps/RenderKind.tsx and reps/RenderedValue.tsx and the currently used one is, i think, in reps/ValueSkeletonWidget.tsx
